### PR TITLE
feat(AutoComplete): AutoComplete support rootClassName

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | AutoComplete support rootClassName |
| 🇨🇳 Chinese | AutoComplete 支持 rootClassName  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 7ba2971</samp>

This pull request adds a new prop `rootClassName` to the `AutoComplete` component, which enables users to customize the class name of the popup element. It also updates the tests, demos and documentation to reflect the new prop.

### <samp>🤖 Generated by Copilot at 7ba2971</samp>

*  Add `rootClassName` prop to `AutoComplete` component to allow customizing the root element and the popup element of the component ([link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dR38), [link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dR64), [link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dL148-R150))
*  Update the documentation of the `AutoComplete` component in both English and Chinese to include the `rootClassName` prop and its details ([link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-f04508ef988a95e022e0488be2b02506610901732efa4dfba77f7ac6e1ae65d1R44), [link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-54fefadfae27caa4550cb59eb92b84d4a0ec3b9b17bb56f9ec3cae2948886db3R45))
*  Add the `rootClassName` prop to the basic demo of the `AutoComplete` component to show its usage and effect ([link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-bc1bb6d6c8968e1095462aaa92e45087a7e402b5ec14981092ede93c80e73b14R36))
*  Add a test case for the `rootClassName` prop of the `AutoComplete` component to verify that the custom root class name is applied to the element ([link](https://github.com/ant-design/ant-design/pull/44055/files?diff=unified&w=0#diff-280a35bf9505afb3855621cda2722a2aae90b25ec454cd1a17ed3963a6714c19R119-R126))

### CODE REVIEW

File: components/auto-complete/demo/basic.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ const App: React.FC = () => {
       <br />
       <br />
       <AutoComplete
+        rootClassName="custom-root-class"
```
Comment (zombieJ) [inline components/auto-complete/demo/basic.tsx:36]: 不要在 demo 里加 root 相关属性

Comment (kiner-tang) [inline components/auto-complete/demo/basic.tsx:36]:
```suggestion

```

File: components/auto-complete/index.en-US.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ The differences with Select are:
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
+| rootClassName | ClassName on the root element | string | - | 5.9.0 |
```
Comment (kiner-tang) [inline components/auto-complete/index.en-US.md:44]:
```suggestion

```

File: components/auto-complete/index.zh-CN.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ demo:
 
 | 参数 | 说明 | 类型 | 默认值 | 版本 |
 | --- | --- | --- | --- | --- |
+| rootClassName | 添加在组件最外层的 className | string | - | 5.9.0 |
```
Comment (kiner-tang) [inline components/auto-complete/index.zh-CN.md:45]:
```suggestion

```

File: components/auto-complete/index.tsx [hunk 4]
```diff
@@ -35,6 +35,7 @@ export interface AutoCompleteProps<
     InternalSelectProps<ValueType, OptionType>,
     'loading' | 'mode' | 'optionLabelProp' | 'labelInValue'
   > {
+  rootClassName?: string;
```
Comment (zombieJ) [inline components/auto-complete/index.tsx:38]: 感觉 `rootClassName` 应该挪到 InternalSelectProps 里，免得两边都重复定义一次。

File: components/auto-complete/index.tsx [hunk 5]
```diff
@@ -145,7 +146,7 @@ const AutoComplete: React.ForwardRefRenderFunction<RefSelectProps, AutoCompleteP
       suffixIcon={null}
       {...omit(props, ['dataSource', 'dropdownClassName'])}
       prefixCls={prefixCls}
-      popupClassName={popupClassName || dropdownClassName}
+      popupClassName={classNames(popupClassName || dropdownClassName)}
```
Comment (zombieJ) [inline components/auto-complete/index.tsx:149]: Select 本身已经支持 `rootClassName` 了，这边不用的

**Review by zombieJ — APPROVED** [review]

**Pull request merged**